c7n/resources/account.py

```diff
@@ -212,6 +221,54 @@ def process(self, resources, event=None):
                         'LatestDeliveryError'):
                     running.append(t)
             trails = running
+        if self.data.get('include-management-events'):
+            matched = []
+            for t in list(trails):
+                selectors = client.get_event_selectors(TrailName=t['TrailARN'])
+                if 'EventSelectors' in selectors.keys():
+                    for s in selectors['EventSelectors']:
+                        if s['IncludeManagementEvents'] and s['ReadWriteType'] == 'All':
+                            matched.append(t)
+            trails = matched
+        if self.data.get('log-metric-filter-pattern'):
+            client_logs = session.client('logs')
+            client_cw = session.client('cloudwatch')
+            sns_manager = self.manager.get_resource_manager('sns-subscription')
+            matched = []
+            for t in list(trails):
+                if 'CloudWatchLogsLogGroupArn' not in t.keys():
+                    continue
+                log_group_name = t['CloudWatchLogsLogGroupArn'].split(':')[6]
+                try:
+                    metric_filters_log_group = \
+                        client_logs.describe_metric_filters(
+                            logGroupName=log_group_name)['metricFilters']
+                except ClientError as e:
+                    if e.response['Error']['Code'] == 'ResourceNotFoundException':
+                        continue
+                filter_matched = None
+                if metric_filters_log_group:
+                    for f in metric_filters_log_group:
+                        if f['filterPattern'] == self.data.get('log-metric-filter-pattern'):
+                            filter_matched = f
+                            break
+                if not filter_matched:
+                    continue
+                alarms = client_cw.describe_alarms_for_metric(
+                    MetricName=filter_matched["metricTransformations"][0]["metricName"],
+                    Namespace=filter_matched["metricTransformations"][0]["metricNamespace"]
+                )['MetricAlarms']
+                alarm_actions = []
+                for a in alarms:
+                    alarm_actions.extend(a['AlarmActions'])
+                if not alarm_actions:
+                    continue
+                alarm_actions = set(alarm_actions)
+                sns_subscriptions = sns_manager.resources()
```

Comment:
@thisisshi fwiw, this line is problematic, versus doing the inverse, ie high cardinality subscription, low cardinality trail, instead of doing a nested loop with high to low, do a low to low.

Reply:
I've refactored it here: https://github.com/cloud-custodian/cloud-custodian/pull/8020 this should be much bet